frontend_tests/node_tests/compose_validate.js

```diff
@@ -460,72 +519,69 @@ test_ui("test_validate_stream_message_post_policy_full_members_only", () => {
     compose_state.topic("subject103");
     compose_state.stream_name("stream103");
     stream_data.add_sub(sub);
+    let banner_rendered = false;
+    mock_template("compose_banner/compose_banner.hbs", false, (data) => {
+        assert.equal(data.classname, compose_banner.CLASSNAMES.no_post_permissions);
+        assert.equal(
+            data.banner_text,
+            $t({
+                defaultMessage: "You do not have permission to post in this stream.",
+            }),
+        );
+        banner_rendered = true;
+        return "post_permissions_stub";
+    });
     assert.ok(!compose_validate.validate());
-    assert.equal(
-        $("#compose-error-msg").html(),
-        $t_html({defaultMessage: "You do not have permission to post in this stream."}),
-    );
+    assert.ok(banner_rendered);
 });
 
-test_ui("test_check_overflow_text", () => {
+test_ui("test_check_overflow_text", ({mock_template}) => {
+    mock_banners();
     page_params.max_message_length = 10000;
 
     const $textarea = $("#compose-textarea");
     const $indicator = $("#compose_limit_indicator");
     const $send_button = $("#compose-send-button");
+    let banner_rendered = false;
+    mock_template("compose_banner/compose_banner.hbs", false, (data) => {
+        assert.equal(data.classname, compose_banner.CLASSNAMES.message_too_long);
+        assert.equal(
+            data.banner_text,
+            $t({
+                defaultMessage: "Message length shouldn't be greater than 10000 characters.",
+            }),
+        );
+        banner_rendered = true;
+        return "overflow_stub";
+    });
 
     // Indicator should show red colored text
     $textarea.val("a".repeat(10000 + 1));
     compose_validate.check_overflow_text();
     assert.ok($indicator.hasClass("over_limit"));
     assert.equal($indicator.text(), "10001/10000");
     assert.ok($textarea.hasClass("over_limit"));
-    assert.equal(
-        $("#compose-error-msg").html(),
-        "translated HTML: Message length shouldn't be greater than 10000 characters.",
-    );
+    assert.ok(banner_rendered);
     assert.ok($send_button.prop("disabled"));
 
-    $("#compose-send-status").stop = () => ({fadeOut: () => {}});
-
     // Indicator should show orange colored text
+    banner_rendered = false;
     $textarea.val("a".repeat(9000 + 1));
     compose_validate.check_overflow_text();
     assert.ok(!$indicator.hasClass("over_limit"));
     assert.equal($indicator.text(), "9001/10000");
     assert.ok(!$textarea.hasClass("over_limit"));
     assert.ok(!$send_button.prop("disabled"));
+    assert.ok(!banner_rendered);
 
     // Indicator must be empty
+    banner_rendered = false;
     $textarea.val("a".repeat(9000));
     compose_validate.check_overflow_text();
     assert.ok(!$indicator.hasClass("over_limit"));
     assert.equal($indicator.text(), "");
     assert.ok(!$textarea.hasClass("over_limit"));
-});
-
-test_ui("test_message_overflow", () => {
```

Comment:
I couldn't see how this test tested anything relevant that the above test wasn't already testing.

Reply:
Seems reasonable.